test/e2e-tests/support/kustomize.go

```diff
@@ -0,0 +1,111 @@
+package support
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	pipepinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+func ResourceToType[T any](resource *resource.Resource, t *T) error {
+	bytes, err := resource.MarshalJSON()
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(bytes, &t)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func KustomizeBuild(path string) (resmap.ResMap, error) {
+	options := krusty.MakeDefaultOptions()
+	k := krusty.MakeKustomizer(options)
+	fs := filesys.FileSystemOrOnDisk{
+		FileSystem: nil,
+	}
+
+	resourceMap, err := k.Run(fs, path)
+	if err != nil {
+		return nil, err
+	}
+
+	return resourceMap, nil
+}
+
+func CreateObjectsFromResourceMap(ctx context.Context, clients *Clients, resourceMap resmap.ResMap, namespace string) error {
+	for _, rsc := range resourceMap.Resources() {
+		kind := rsc.GetKind()
+		switch kind {
+		case "ConfigMap":
+			{
+				var configMap corev1.ConfigMap
+				err := ResourceToType(rsc, &configMap)
+				if err != nil {
+					return err
+				}
+
+				_, err = clients.Kubernetes.CoreV1().ConfigMaps(namespace).Get(ctx, configMap.Namespace, metav1.GetOptions{})
```

Comment:
I didn't notice it until the changes to parameterise the namespace, but it seems odd that the namespace is needed to create the `Clients` instance, and then is still also needed for functions that are based on that. My feeling is that our Clients abstraction could be improved in one direction or the other.